src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -924,6 +926,105 @@ namespace OpenMS
     }
   }
 
+
+  void TheoreticalSpectrumGenerator::getPrefixAndSuffixIonsMZ(std::vector<float>& spectrum, const AASequence& peptide, int charge) const
+  {
+    for (Int z = charge; z >= 1; --z)
+    {
+      if (add_b_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::BIon, z);
+      }
+      if (add_y_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::YIon, z);
+      }
+      if (add_a_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::AIon,  z );
+      }
+      if (add_x_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::XIon, z);
+      }
+      if (add_c_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::CIon, z);
+      }
+      if (add_z_ions_)
+      {
+        addPrefixAndSuffixIons_(spectrum, peptide, Residue::ZIon, z);
+      }
+    }
+
+
+    std::sort(spectrum.begin(), spectrum.end());
+
+
+    return;
+  }
+
+  void TheoreticalSpectrumGenerator::addPrefixAndSuffixIons_(std::vector<float>& spectrum, const OpenMS::AASequence& peptide, Residue::ResidueType res_type, int charge)
+  {
+    if (peptide.empty())
+    {
+      cout << "Warning: Attempt at creating Prefix and Suffix Ions Spectrum from empty string!" << endl;
+      return;
+    }
+
+    if (res_type == Residue::AIon || res_type == Residue::BIon || res_type == Residue::CIon)
+    {
+      double mono_weight(Constants::PROTON_MASS_U * charge);
+      if (peptide.hasNTerminalModification())
+      {
+        mono_weight += peptide.getNTerminalModification()->getDiffMonoMass();
+      }
+
+      switch (res_type)
+      {
+        case Residue::AIon: mono_weight += Residue::getInternalToAIon().getMonoWeight(); break;
```

Comment:
@jpfeuffer I vaguely recall that you changed something in the TSG regarding the calculation of internal masses. Does this code also need this update?

Reply:
I think it was not merged yet

Reply:
Let em check... I think https://github.com/OpenMS/OpenMS/pull/7226/files was merged through https://github.com/OpenMS/OpenMS/pull/7237/files just with some additional fixes. Was there another one?

Reply:
ah it was here. https://github.com/OpenMS/OpenMS/pull/7237/files#diff-11be5715633a0e04aa066f79aa957d602ed73dc63e9d9fb2239d270356ede638R414 so it already got merged... all good